### PR TITLE
Adding scaling to modify_demand.py

### PR DESCRIPTION
**Pull request by alfiyaks**
Added scaling based on snapshots and "load_options":"scale" parameter in config file.

### CODE REVIEW

File: scripts/modify_demand.py [hunk 1]
```diff
@@ -62,8 +63,10 @@ def rescale_load(n, gadm_demand):
     
     # Snakemake imports:
     gadm_demand = snakemake.input["gadm_demand_data"]
-    
-    rescale_load(n, gadm_demand)
+    start_date = datetime.strptime(snakemake.config["snapshots"]["start"], "%Y-%m-%d")
+    end_date =datetime.strptime( snakemake.config["snapshots"]["end"], "%Y-%m-%d")
+    scale = snakemake.config["load_options"]["scale"] *(end_date-start_date).days/365
```
Comment (martacki) [inline scripts/modify_demand.py:68]: If you want to model more or less than 1 yr, why should the demand of the modelled timeframe be scaled by a fraction of the year?
Assuming we model January 2013 only, then the January demand will be reduced by approx. 91%. I guess I am missing something here?

Comment (alfiyaks) [reply scripts/modify_demand.py:68]: In rule `def rescale_load(n, gadm_demand):` the `scale_factor` is taken as a yearly regional demand `official_demand` divided by total demand in pypsa network `pypsa_demand`. 
`scale_factor = (official_demand["Consumption (GWh)"] * 1e3) / pypsa_demand`
If pypsa network was modelled for a fraction of a year, e.g. 1 month, our scaling will be  `yearly_demand/monthly_demand`, which will scale all our regional demands to a yearly consumption, no matter we are simulating one week, one year or ten years. So we need to rescale to take into account snapshots to have accurate representation.

On the picture below you can see what happened when I tried to simulate 1 week.

image

**Review by martacki — APPROVED** [review]
Ah I see, thank you for clarification. Yes, this makes sense. Thank you for carefully checking and contributing back to our main source code!

LGTM.

**Comment by alfiyaks** [comment]
Thanks, Marta!

**Pull request merged**